fenix/app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuExperimentsRobot.kt

```diff
@@ -0,0 +1,63 @@
+@file:Suppress("TooManyFunctions")
+
+package org.mozilla.fenix.ui.robots
+
+import android.view.View
+import android.widget.TextView
+import androidx.compose.ui.platform.ComposeView
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.matcher.ViewMatchers.withContentDescription
+import androidx.test.uiautomator.By.textContains
+import androidx.test.uiautomator.UiSelector
+import org.junit.Assert.assertTrue
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTimeShort
+import org.mozilla.fenix.helpers.TestHelper.mDevice
+import org.mozilla.fenix.helpers.click
+
+/**
+ * Implementation of Robot Pattern for the experiments sub menu.
+ */
+class SettingsSubMenuExperimentsRobot {
+
+    class Transition {
+
+        fun goBackToHomeScreen(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            goBackButton().click()
+
+            HomeScreenRobot().interact()
+            return HomeScreenRobot.Transition()
+        }
+
+        fun goBack(interact: SettingsRobot.() -> Unit): SettingsRobot.Transition {
+            goBackButton().click()
+
+            SettingsRobot().interact()
+            return SettingsRobot.Transition()
+        }
+
+        fun clickSnackBarViewButton(interact: HomeScreenRobot.() -> Unit): HomeScreenRobot.Transition {
+            val snackBarButton = mDevice.findObject(UiSelector().text("VIEW"))
+            snackBarButton.waitForExists(waitingTimeShort)
+            snackBarButton.click()
+
+            HomeScreenRobot().interact()
+            return HomeScreenRobot.Transition()
```

Comment:
Oh yeah, if `goBackToHomeScreen` `goBack` `clickSnackBarViewButton` are not called in your test, feel free to remove them too.

Reply:
I'll remove ```clickSnackBarViewButton ```, the other 2 will be useful in the future.